idris-repl.el

```diff
@@ -186,9 +186,12 @@ If ALWAYS-INSERT is non-nil, always insert a prompt at the end of the buffer."
       (idris-repl-insert-prompt)
       (insert current-input))))
 
+(autoload 'idris-load-file-sync "idris-commands.el")
+;;;###autoload
 (defun idris-switch-to-repl ()
-  "Select the output buffer and scroll to bottom."
-  (interactive)
+  "Load the current Idris file buffer and jump to the Idris REPL."
+  (interactive nil idris-mode)
```

Comment:
Specifying `idris mode` should exclude this command from M-x list in non idris-mode based buffers. Depends on user setting as per documentation:

```
If MODES is present, it should be one or more mode names (symbols)
for which this command is applicable.  This is so that ‘M-x TAB’
will be able to exclude this command from the list of completion
candidates if the current buffer’s mode doesn’t match the list.
Which commands are excluded from the list of completion
candidates based on this information is controlled by the value
of ‘read-extended-command-predicate’, which see.
```